### PR TITLE
Update version and add Date header to response

### PR DESCRIPTION
**Pull request by wowi42**
### Description of the Change

Update a dependency and add a Date header to response, in order to respect RFC — https://datatracker.ietf.org/doc/html/rfc2616#section-14.18

### Benefits

Compatible with RFC

### Possible Drawbacks

None

Linked to #668

### CODE REVIEW

File: shard.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: kemal
-version: 1.4.0
+version: 1.4.1
```
Comment (straight-shoota) [inline shard.yml:2]: **chore:** Please do not update the shard version. A new release might happen after this PR is merged, but not as part of it.
```suggestion
version: 1.4.0
```

File: shard.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@ dependencies:
     version: ~> 0.4.0
   exception_page:
     github: crystal-loot/exception_page
-    version: ~> 0.3.0
+    version: ~> 0.4.0
```
Comment (straight-shoota) [inline shard.yml:13]: **chore:** This dependency update seems unrelated to adding a date header, so it should be a separate PR.
```suggestion
    version: ~> 0.3.0
```

File: src/kemal/init_handler.cr [hunk 3]
```diff
@@ -9,6 +9,7 @@ module Kemal
     def call(context : HTTP::Server::Context)
       context.response.headers.add "X-Powered-By", "Kemal" if Kemal.config.powered_by_header?
       context.response.content_type = "text/html" unless context.response.headers.has_key?("Content-Type")
+      context.response.headers.add "Date", Time.local.to_rfc2822
```
Comment (straight-shoota) [inline src/kemal/init_handler.cr:12]: **suggestion:** `Time#to_rfc2822` produces a date format in UTC. So there's no local time information involved.
```suggestion
      context.response.headers.add "Date", Time.utc.to_rfc2822
```
`Time.utc` is sufficient. And it's typically more performant than `Time.local`.

**Review by straight-shoota — COMMENTED** [review]
**todo:** It would be nice to have some specs for this.

**Comment by wowi42** [comment]
I will close thins one, and make it cleaner
